sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubUnboundedSink.java

```diff
@@ -242,7 +271,19 @@ private static class WriterFn extends DoFn<KV<Integer, Iterable<OutgoingMessage>
 
     /** BLOCKING Send {@code messages} as a batch to Pubsub. */
     private void publishBatch(List<OutgoingMessage> messages, int bytes) throws IOException {
-      int n = pubsubClient.publish(topic.get(), messages);
+      int n = 0;
+      if (topic != null) {
+        n = pubsubClient.publish(topic.get(), messages);
+      } else {
+        Map<TopicPath, List<OutgoingMessage>> messagesPerTopic = Maps.newHashMap();
+        for (OutgoingMessage message : messages) {
+          TopicPath topicPath = PubsubClient.topicPathFromPath(message.topic());
```

Comment:
could do this translation during publish loop so it's per topic instead of per message

Reply:
done